### PR TITLE
New walls, floors and windows

### PR DESCRIPTION
**Pull request by CakeQ**
Replaces walls, floors and windows with my own sprites. 

Also makes WindowComponent a child of IconSmoothComponent so that it can be more flexible, as well as gave windows a non-framed sprite set. Allows for windows with colors now.

~~Modifies Saltern station with a vague flooring layout~~ Will map in flooring in a seperate PR

Pics:
image

image

image

image

image

image

### CODE REVIEW

**Comment by SweptWasTaken** [comment]
Hmm.

For what purpose?

**Comment by CakeQ** [comment]
> 
> 
> Hmm.
> 
> For what purpose?

because I wanted to port my turf sprites into ss14

and also because I intend on porting my airlocks as well as further working on the turf set to portray 3/4 view better

As the one who created the turf set I believe there is a stronger coherency across the assets and the work well together, though I'm aware that's just my opinion

**Comment by CakeQ** [comment]
Airlock sprites for comparison:

**Comment by AJCM-git** [comment]
I like it, the only criticism i have so far is that the windows dont merge and that the floor is too dark. Maybe keep Eris floor sprites?

**Comment by Hugal31** [comment]
I'm not sure I like profile angle for the airlocks

**Comment by AJCM-git** [comment]
Thats the nostalgia talking, perspective airlocks are superior

**Comment by SweptWasTaken** [comment]
Content Client_2020-05-27_11-15-43
Content Client_2020-05-27_11-16-10
Content Client_2020-05-27_11-20-40

**Comment by CakeQ** [comment]
more progress to come

**Comment by CakeQ** [comment]
Updated PR based off of feedback, as well as implemented the rest of the wall layering.

Added two new paint layers to walls for better detail, as well as edge pieces that connect between windows, walls, and airlocks.

Updated previews:
image

image

image

Some further closeups:

image

image

image

**Comment by CakeQ** [comment]
Updated colors cause PJB wasn't happy
image

image

**Comment by ComicIronic** [comment]
I agree that the lighter wall colours look better.

**Comment by CakeQ** [comment]
Forgot to mention that I also added wooden walls: 
image

**Comment by CakeQ** [comment]
PR no longer requires engine changes

File: Content.Client/GameObjects/Components/IconSmoothing/IconSmoothComponent.cs [hunk 1]
```diff
@@ -41,13 +44,19 @@ public class IconSmoothComponent : Component
         /// <summary>
         ///     We will smooth with other objects with the same key.
         /// </summary>
+        [ViewVariables(VVAccess.ReadWrite)]
```
Comment (ComicIronic) [inline Content.Client/GameObjects/Components/IconSmoothing/IconSmoothComponent.cs:47]: Should be readonly.

File: Content.Client/GameObjects/Components/WindowComponent.cs [hunk 2]
```diff
@@ -1,79 +1,141 @@
...
-        public void UpdateSprite()
+        [Pure]
+        private bool MatchingWall(IEnumerable<IEntity> candidates)
         {
-            var lowWall = FindLowWall();
-            if (lowWall == null)
+            foreach (var entity in candidates)
             {
-                return;
+                if (entity.TryGetComponent(out IconSmoothComponent other))
```
Comment (ComicIronic) [inline Content.Client/GameObjects/Components/WindowComponent.cs:122]: Should this not be negated?

**Comment by ComicIronic** [comment]
All in all, I think it's a real visual improvement.

File: Content.Client/GameObjects/Components/LowWallComponent.cs [hunk 3]
```diff
@@ -33,6 +35,20 @@ public class LowWallComponent : IconSmoothComponent
...
+        private string _overlayColor;
+
+        [ViewVariables(VVAccess.ReadWrite)]
+        private string _edgeColor;
+
+        public override void ExposeData(ObjectSerializer serializer)
+        {
+            base.ExposeData(serializer);
+
+            serializer.DataField(ref _overlayColor, "overlayColor", null);
```
Comment (ComicIronic) [inline Content.Client/GameObjects/Components/LowWallComponent.cs:48]: I *believe* the serializer will do to/from hex for you automatically if you give it a field which has type `Color`, so it should be possible to make your `string` fields `Color`s instead and cut out `FromHex` everywhere you've used it.

Comment (CakeQ) [reply Content.Client/GameObjects/Components/LowWallComponent.cs:48]: Works

File: Content.Client/GameObjects/Components/WindowComponent.cs (hunk 2 above)
Comment (CakeQ) [reply Content.Client/GameObjects/Components/WindowComponent.cs:122]: It should

**Comment by PJB3005** [comment]
This is completely a subjective change so I'm gonna leave this one up to the emojiocracy on the main PR.

**Comment by SweptWasTaken** [comment]
I personally don't like these walls replacing our current walls. 

They look good no doubt but in my opinion our old ones look better

If this pr doesn't get passed we could still use these walls at Nanotransen or somewhere...

**Pull request closed without merging**